### DISCUSSION TITLE
Is there a way to send arguments to the simulator?

### DISCUSSION THREAD

**Discussion by sei-jgwohlbier** [fastmachinelearning/hls4ml#1114] [Q&A] [answered]
I'm using the Vitis backend. Is there a way to send arguments to the invocation of the simulator besides manually editing the `run_xsim.sh` file that is generated?

**Comment by bo3z** [comment]
You can specify input_data_tb and output_data_tb as NumPy arrays or .dat files. See here for Keras: https://github.com/fastmachinelearning/hls4ml/blob/2c17f66002791c602e5295c7f6f678601575eb31/hls4ml/converters/__init__.py#L181. Analogous for PyTorch.

This is then picked up and forwarded to both CSim (software simulation) and CoSim (RTL simulation)

**Reply by sei-jgwohlbier** [reply]
I was thinking about arguments to the `xsim` command itself, like how I added `--maxdeltaid 100000 --tp --tl` manually below to `run_xsim.sh`.
```
/tools/Xilinx/Vivado/2023.2/bin/xsim --maxdeltaid 100000 --tp --tl --noieeewarnings myproject -tclbatch myproject.tcl -view myproject_dataflow_ana.wcfg -protoinst myproject.protoinst
```

**Reply by bo3z** [reply, marked as answer]
Unfortunately, these are not directly supported in hls4ml and need to be added manually.
